src/main/scala/Parser.scala

```diff
@@ -261,8 +261,8 @@ private class FuseParser extends RegexParsers with PackratParsers {
     }
   }
   lazy val funcDef: P[FuncDef] = positioned {
-    "def" ~> iden ~ parens(repsep(args, ",")) ~ retTyp ~ block ^^ {
-      case fn ~ args ~ ret ~ body => FuncDef(fn, args, ret, Some(body))
+    "def" ~> iden ~ parens(repsep(args, ",")) ~ retTyp ~ "=" ~ block ^^ {
```

Comment:
Any particular reason for this syntax change, adding a `=` to function definitions?

Reply:
Turns out the grammar is ambiguous/requires a lot of back tracking without it. This is because code blocks have the same syntax:

```
let x = 1;
{
  let y = 2 + x;
}
```